src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final int INDEX_SUPPLY = 0;
+    public static final int INDEX_BUY = 1;
+    public static final int INDEX_SUM_ARRAY_LENGTH = 2;
+    public static final int ARRAY_COUNT = 1;
+    public static final String SUPPLY = "supply";
+    public static final String BUY = "buy";
+    public static final String RESULT = "result";
+    public static final String DELIMITER = "\\W+";
+    public static final String COMMA = ",";
+    public static final String SEPARATOR = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeList(createReport(readList(fromFileName)),toFileName);
+    }
+
+    private String readList(String fromFileName) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            int value = reader.read();
+            while (value != -1) {
+                stringBuilder.append((char) value);
+                value = reader.read();
+            }
```

Comment:
you can read the entire line instead of a every single byte.
```suggestion
           String line;
            while ((line = reader.readLine()) != null) {
                stringBuilder.append(line);
            }
```